### PR TITLE
fix(messages): show past messages in correct order

### LINKED ISSUE #1024: bug(chat): Messages are not coming in, in order
**Issue:** 
If you notice when you log into the app and scroll through some of your chats the messages dont always appear as they should. Example in the video shows some of the messages Jeff and I sent at the bottom there are messages sent from 2 days ago, then in the middle it says 3 days ago, and in the very top its back to 2 days ago. Im also attaching the video Sheldon had as well 

**OS:**
All

**Screen recording:**

https://github.com/Satellite-im/Uplink/assets/93608357/e660a4d6-776c-404a-9ed4-eefbae6405e1

https://github.com/Satellite-im/Uplink/assets/93608357/e23f5b0e-22a5-4112-8826-b78534281584

### PR DESCRIPTION
**Pull request by sdwoodbury**
### What this PR does 📖

- fixes the lazy loading of messages. Chats are still only initialized with a few messages at first but now as more messages are fetched, they are displayed in order. 
- This fix isn't as lazy as it could be, but doing it this way prevents bugs arising from extra events getting triggered. 
- This may be good for a temporary solution, until we figure out how to determine when the user has scrolled to the end of a container. Without that, infinite scrolling will be hacky at best. 

### Which issue(s) this PR fixes 🔨

- Resolve #1024

### CODE REVIEW

**Review by dariusc93 — APPROVED** [review]
LGTM, though it makes me wonder if we should switch over to using `MessagePage` instead and as one scroll up (or down from the current page of messages) that it would get the next (or previous) set.

**Comment by sdwoodbury** [comment]
> LGTM, though it makes me wonder if we should switch over to using `MessagePage` instead and as one scroll up (or down from the current page of messages) that it would get the next (or previous) set.

I'd like that. I'm hoping we could use an approach like that once I figure out how to handle scrolling in Dioxus.

**Review by stavares843 — CHANGES_REQUESTED** [review]
ignore

**Review by stavares843 — APPROVED** [review]

File: common/src/warp_runner/ui_adapter/mod.rs [hunk 1]
```diff
@@ -154,38 +154,52 @@ pub async fn dids_to_identity(
...
     .await;
 
-    Ok(messages)
+    let has_more = to_skip > 0;
+    // log::debug!(
+    //     "fetched messages. total: {}, num taken: {}, has_more: {}",
+    //     total_messages,
+    //     to_take,
+    //     has_more
+    // );
```
Comment (stavares843) [inline common/src/warp_runner/ui_adapter/mod.rs:185]: do we need this part? 🔨

**Review by phillsatellite — APPROVED** [review]

**Comment by stavares843** [comment]
added a minor comment but merging

**Pull request merged**